agent/structs/config_entry_mesh.go

```diff
@@ -20,6 +20,11 @@ type MeshConfigEntry struct {
 	// MutualTLSMode=permissive in either service-defaults or proxy-defaults.
 	AllowEnablingPermissiveMutualTLS bool `json:",omitempty" alias:"allow_enabling_permissive_mutual_tls"`
 
+	// ValidateClusters is false by default and configures whether Envoy proxies will validate clusters in a route. If
+	// set to true and any clusters in the route do not exist, the route table will not load. If set to false, the route
+	// table will load and routing to a non-existent cluster will result in a 404.
```

Comment:
~ We might want to add a link the Envoy docs for this option as well
```suggestion
	// table will load and routing to a non-existent cluster will result in a 404.
	// See [Envoy docs] for more details.
	//
	// [Envoy docs]: https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route.proto#envoy-v3-api-field-config-route-v3-routeconfiguration-validate-clusters
```